lib/graphql/schema/build_from_ast.rb

```diff
@@ -0,0 +1,260 @@
+          when GraphQL::Language::Nodes::Enum
+            input_argument.default_value.name
+          else
+            input_argument.default_value
+          end
+
+          [
+            input_argument.name,
+            GraphQL::Argument.define(
+              name: input_argument.name,
+              type: type_resolver.call(input_argument.type),
+              description: input_argument.description,
+              default_value: default_value,
+            )
+          ]
+        end
+      end
+
+      def build_directive(directive_definition, type_resolver)
+        GraphQL::Directive.define(
+          name: directive_definition.name,
+          description: directive_definition.description,
+          arguments: Hash[build_directive_arguments(directive_definition, type_resolver)],
+          locations: directive_definition.locations.map(&:to_sym),
+        )
+      end
+
+      def build_directive_arguments(directive_definition, type_resolver)
+        directive_definition.arguments.map do |directive_argument|
+          [
+            directive_argument.name,
+            GraphQL::Argument.define(
+              name: directive_argument.name,
+              type: type_resolver.call(directive_argument.type),
+              description: directive_argument.description,
+              default_value: directive_argument.default_value,
+            )
+          ]
+        end
+      end
+
+      def build_interface_type(interface_type_definition, type_resolver)
+        GraphQL::InterfaceType.define(
+          name: interface_type_definition.name,
+          description: interface_type_definition.description,
+          fields: Hash[build_fields(interface_type_definition.fields, type_resolver)],
+        )
+      end
+
+      def build_fields(field_definitions, type_resolver)
+        field_definitions.map do |field_definition|
+          field_arguments = Hash[field_definition.arguments.map do |argument|
+            default_value = case argument.default_value
+            when GraphQL::Language::Nodes::Enum
+              argument.default_value.name
+            else
+              argument.default_value
+            end
+
+            arg = GraphQL::Argument.define(
+              name: argument.name,
+              description: argument.description,
+              type: type_resolver.call(argument.type),
+              default_value: default_value,
+            )
+
+            [argument.name, arg]
+          end]
+
+          [
+            field_definition.name,
+            GraphQL::Field.define(
+              name: field_definition.name,
+              description: field_definition.description,
+              type: type_resolver.call(field_definition.type),
+              arguments: field_arguments,
+              deprecation_reason: build_deprecation_reason(field_definition.directives),
+            )
+          ]
+        end
+      end
+
+      def resolve_type(types, ast_node)
```

Comment:
I was considering refactoring [`Schema::TypeExpression.build_type`](https://github.com/rmosolgo/graphql-ruby/blob/c78d24bb1f2fbfa5b0e486c4504d2ecc334cb700/lib/graphql/schema/type_expression.rb) to accept a hash of `types` instead of `schema`.

This would allow us to :fire: this entire method.